builtInFunctions/esdtMetaDataUpdate.go

```diff
@@ -0,0 +1,158 @@
+package builtInFunctions
+
+import (
+	"fmt"
+	"math/big"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type esdtMetaDataUpdate struct {
+	baseActiveHandler
+	funcGasCost           uint64
+	globalSettingsHandler vmcommon.GlobalMetadataHandler
+	storageHandler        vmcommon.ESDTNFTStorageHandler
+	rolesHandler          vmcommon.ESDTRoleHandler
+	accounts              vmcommon.AccountsAdapter
+	enableEpochsHandler   vmcommon.EnableEpochsHandler
+	gasConfig             vmcommon.BaseOperationCost
+	mutExecution          sync.RWMutex
+}
+
+// NewESDTMetaDataUpdateFunc returns the esdt meta data update built-in function component
+func NewESDTMetaDataUpdateFunc(
+	funcGasCost uint64,
+	gasConfig vmcommon.BaseOperationCost,
+	accounts vmcommon.AccountsAdapter,
+	globalSettingsHandler vmcommon.GlobalMetadataHandler,
+	storageHandler vmcommon.ESDTNFTStorageHandler,
+	rolesHandler vmcommon.ESDTRoleHandler,
+	enableEpochsHandler vmcommon.EnableEpochsHandler,
+) (*esdtMetaDataUpdate, error) {
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(globalSettingsHandler) {
+		return nil, ErrNilGlobalSettingsHandler
+	}
+	if check.IfNil(enableEpochsHandler) {
+		return nil, ErrNilEnableEpochsHandler
+	}
+	if check.IfNil(storageHandler) {
+		return nil, ErrNilESDTNFTStorageHandler
+	}
+	if check.IfNil(rolesHandler) {
+		return nil, ErrNilRolesHandler
+	}
+
+	e := &esdtMetaDataUpdate{
+		accounts:              accounts,
+		globalSettingsHandler: globalSettingsHandler,
+		storageHandler:        storageHandler,
+		rolesHandler:          rolesHandler,
+		enableEpochsHandler:   enableEpochsHandler,
+		funcGasCost:           funcGasCost,
+		gasConfig:             gasConfig,
+		mutExecution:          sync.RWMutex{},
+	}
+
+	e.baseActiveHandler.activeHandler = enableEpochsHandler.IsDynamicESDTEnabled
+
+	return e, nil
+}
+
+// ProcessBuiltinFunction saves the token type in the system account
+func (e *esdtMetaDataUpdate) ProcessBuiltinFunction(acntSnd, _ vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	err := checkArguments(vmInput, acntSnd, e.baseActiveHandler)
+	if err != nil {
+		return nil, err
+	}
+	if len(vmInput.Arguments) < 7 {
+		return nil, ErrInvalidNumberOfArguments
+	}
+
+	err = e.rolesHandler.CheckAllowedToExecute(acntSnd, vmInput.Arguments[tokenIDIndex], []byte(core.ESDTRoleNFTUpdate))
+	if err != nil {
+		return nil, err
+	}
+
+	totalLength := uint64(0)
+	for _, arg := range vmInput.Arguments {
+		totalLength += uint64(len(arg))
+	}
+
+	e.mutExecution.RLock()
+	gasToUse := totalLength*e.gasConfig.StorePerByte + e.funcGasCost
```

Comment:
take storePerByte only for the difference.

Reply:
Done.